staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -60,7 +60,11 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 	}
 	// The options didn't require a custom TLS config
 	if tlsConfig == nil {
-		return http.DefaultTransport, nil
+		tr := http.DefaultTransport
+		if defaultTransport, ok := tr.(*http.Transport); ok {
+			defaultTransport.DisableCompression = config.DisableCompression
```

Comment:
this is changing the DisableCompression setting in the shared `http.DefaultTransport`, which affects all users of that transport (and possibly `http.DefaultClient`)